src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestExtractFileValidationServiceImpl.java

```diff
@@ -3,28 +3,169 @@
+import org.kuali.rice.kim.api.identity.Person;
+import org.kuali.rice.kim.api.identity.PersonService;
+
+import edu.cornell.kfs.concur.ConcurConstants;
+import edu.cornell.kfs.concur.ConcurParameterConstants;
+import edu.cornell.kfs.concur.batch.ConcurRequestExtractPdpConstants;
+import edu.cornell.kfs.concur.batch.businessobject.ConcurRequestedCashAdvance;
 import edu.cornell.kfs.concur.batch.businessobject.ConcurRequestExtractFile;
 import edu.cornell.kfs.concur.batch.businessobject.ConcurRequestExtractRequestDetailFileLine;
 import edu.cornell.kfs.concur.batch.businessobject.ConcurRequestExtractRequestEntryDetailFileLine;
 import edu.cornell.kfs.concur.batch.service.ConcurRequestExtractFileValidationService;
-
-import org.kuali.rice.core.api.util.type.KualiDecimal;
+import edu.cornell.kfs.concur.batch.service.ConcurRequestedCashAdvanceService;
+import edu.cornell.kfs.concur.businessobjects.ConcurAccountInfo;
+import edu.cornell.kfs.concur.businessobjects.ValidationResult;
+import edu.cornell.kfs.concur.service.ConcurAccountValidationService;
+import edu.cornell.kfs.sys.CUKFSConstants;
+import edu.cornell.kfs.sys.CUKFSParameterKeyConstants;
 
 public class ConcurRequestExtractFileValidationServiceImpl implements ConcurRequestExtractFileValidationService {
-
     private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(ConcurRequestExtractFileValidationServiceImpl.class);
+    protected ConcurRequestedCashAdvanceService concurRequestedCashAdvanceService;
+    protected ConcurAccountValidationService concurAccountValidationService;
+    protected ParameterService parameterService;
+    protected PersonService personService;
 
-    public boolean requestExtractHeaderRowValidatesToFileContents(List<ConcurRequestExtractFile> requestExtractFiles) {
+    public boolean requestExtractHeaderRowValidatesToFileContents(ConcurRequestExtractFile requestExtractFile) {
         boolean headerValidationPassed = false;
-        if (requestExtractFiles.isEmpty() || requestExtractFiles.size() != 1) {
-            LOG.error("Header validation should have received single parsed file. More or less than one parsed file was detected.");
+        LOG.info("batchDate=" + requestExtractFile.getBatchDate() + "=   recordCount=" + requestExtractFile.getRecordCount() + "=     totalApprovedAmount=" + requestExtractFile.getTotalApprovedAmount() + "=");
+        headerValidationPassed = (fileRowCountMatchesHeaderRowCount(requestExtractFile) &&
+                                  fileOnlyContainsOurEmployeeCustomerIndicator(requestExtractFile) &&
+                                  fileTotalApprovedAmountsMatchHeaderTotalApprovedAmount(requestExtractFile));
+        LOG.info("Header validation : " + ((headerValidationPassed) ? "PASSED" : "FAILED"));
+        return headerValidationPassed;
+    }
+
+    public void performRequestDetailLineValidation(ConcurRequestExtractRequestDetailFileLine detailFileLine) {
```

Comment:
My concern here is that when we call this method to perform a validation and pass in a BO it is not clear that we are also changing the state of that BO.